### PR TITLE
feat: Degrees of separation reference module added

### PR DESCRIPTION
**Pull request by donosonaumczuk**

### CODE REVIEW

File: test/helpers/errors.ts [hunk 1]
```diff
@@ -10,6 +10,12 @@ export const ERRORS = {
   ONLY_OWNER: 'OnlyOwner()',
   INVALID_PARAMETERS: 'InvalidParameters()',
 
+  // DegreesOfSeparationReferenceModule
+  INVALID_DEGREES_OF_SEPARATION: 'InvalidDegreesOfSeparation()',
+  OPEREATION_DISABLED: 'OpereationDisabled()',
```
Comment (bensprks) [inline test/helpers/errors.ts]: Spelling fix

```suggestion
  OPERATION_DISABLED: 'OperationDisabled()',
```

File: contracts/reference/DegreesOfSeparationReferenceModule.sol [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+     * root publication and the profile authoring the comment.
+     */
+    function _validateDegreesOfSeparationRestriction(
+        uint256 profileId,
+        uint256 profileIdPointed,
+        uint8 degreesOfSeparation,
+        uint256[] memory profilePath
+    ) internal view {
+        if (degreesOfSeparation == 0) {
+            revert OpereationDisabled();
```
Comment (bensprks) [inline contracts/reference/DegreesOfSeparationReferenceModule.sol]: Spelling fix
```suggestion
            revert OperationDisabled();
```

File: contracts/reference/DegreesOfSeparationReferenceModule.sol [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+    event ModuleParametersUpdated(
+        uint256 indexed profileId,
+        uint256 indexed pubId,
+        bool commentsRestricted,
+        bool mirrorsRestricted,
+        uint8 degreesOfSeparation
+    );
+
+    error InvalidDegreesOfSeparation();
+    error OpereationDisabled();
```
Comment (bensprks) [inline contracts/reference/DegreesOfSeparationReferenceModule.sol]: Spelling error
```suggestion
    error OperationDisabled();
```

File: test/modules/reference/degrees-of-separation-reference-module.spec.ts [hunk 4]
```diff
@@ -0,0 +1,1068 @@
...
+              contentURI: MOCK_URI,
+              profileIdPointed: ROOT_AUTHOR_PROFILE,
+              pubIdPointed: FIRST_PROFILE_ID,
+              referenceModuleData: abiCoder.encode(['uint256[]'], [[]]),
+              collectModule: freeCollectModule.address,
+              collectModuleInitData: abiCoder.encode(['bool'], [false]),
+              referenceModule: ethers.constants.AddressZero,
+              referenceModuleInitData: [],
+            })
+          ).to.be.revertedWith(ERRORS.OPEREATION_DISABLED);
```
Comment (bensprks) [inline test/modules/reference/degrees-of-separation-reference-module.spec.ts]: Spelling fix
```suggestion
          ).to.be.revertedWith(ERRORS.OPERATION_DISABLED);
```

File: test/modules/reference/degrees-of-separation-reference-module.spec.ts [hunk 5]
```diff
@@ -0,0 +1,1068 @@
...
+          await expect(
+            lensHub.connect(ownerOfFirstDegreeProfile).mirror({
+              profileId: FIRST_DEGREE_PROFILE,
+              profileIdPointed: ROOT_AUTHOR_PROFILE,
+              pubIdPointed: FIRST_PROFILE_ID,
+              referenceModuleData: abiCoder.encode(['uint256[]'], [[]]),
+              referenceModule: ethers.constants.AddressZero,
+              referenceModuleInitData: [],
+            })
+          ).to.be.revertedWith(ERRORS.OPEREATION_DISABLED);
```
Comment (bensprks) [inline test/modules/reference/degrees-of-separation-reference-module.spec.ts]: Spelling fix
```suggestion
          ).to.be.revertedWith(ERRORS.OPERATION_DISABLED);
```

File: test/helpers/errors.ts (hunk 1 above)
Comment (donosonaumczuk) [reply test/helpers/errors.ts]: Done!

File: contracts/reference/DegreesOfSeparationReferenceModule.sol (hunk 2 above)
Comment (donosonaumczuk) [reply contracts/reference/DegreesOfSeparationReferenceModule.sol]: Also Done! Really good catch ser

File: contracts/reference/DegreesOfSeparationReferenceModule.sol (hunk 3 above)
Comment (donosonaumczuk) [reply contracts/reference/DegreesOfSeparationReferenceModule.sol]: Done!

File: test/modules/reference/degrees-of-separation-reference-module.spec.ts (hunk 4 above)
Comment (donosonaumczuk) [reply test/modules/reference/degrees-of-separation-reference-module.spec.ts]: Done!

File: test/modules/reference/degrees-of-separation-reference-module.spec.ts (hunk 5 above)
Comment (donosonaumczuk) [reply test/modules/reference/degrees-of-separation-reference-module.spec.ts]: Done!

**Review by vicnaum — APPROVED** [review]

**Pull request merged**